iree-requirements-ci.txt

```diff
@@ -10,5 +10,5 @@
 # Uncomment to skip versions from PyPI (so _only_ nightly versions).
 # --no-index
 
-iree-base-compiler<=3.0.0
-iree-base-runtime<=3.0.0
+iree-base-compiler<=3.1.0rc20241120
```

Comment:
I want this commit included : https://github.com/iree-org/iree/commit/8fd3e0dd4f213393d89c6818175b67f55940b536

Reply:
Do you know which nightly release this commit would be in?

Reply:
> I want this commit included : [iree-org/iree@8fd3e0d](https://github.com/iree-org/iree/commit/8fd3e0dd4f213393d89c6818175b67f55940b536)
> Do you know which nightly release this commit would be in?

That commit missed the cut last night. It will be included in tonight's nightly pre-release.

However, since that change is not a breaking change and is only changing internal implementation details, you don't _need_ to update the requirements in this PR. All that the CI in this repository enforces is that the builds and tests pass. If you want the CI using a newer pre-release for other reasons (e.g. benchmarking), then you can bump the version.

The packages are decoupled outside of the stable releases. If a user/developer wants to use a newer package, they can just install a newer version of that package. When we release stable versions to pypi (3.0.0, 3.1.0, etc.), the expectation is that the major and minor versions will line up (and the downstream shark-ai meta package enforces that).

Reply:
Make sense, thanks for the explanation. I need that commit for this PR: https://github.com/iree-org/iree-turbine/pull/284 which will fail without it, but I will update the ci in the commit to pick up tonight's pre-release.

Reply:
Gotcha. You can monitor for pre-releases at https://github.com/iree-org/iree/releases .

You can also watch the commit itself to see when it gets associated with a release tag. For example, https://github.com/iree-org/iree/commit/e3826d5eb05e19f5d44ba50de73fd8352cfcb838 was included in all of these releases: `iree-3.1.0rc20241120`, `iree-3.1.0rc20241119`, `iree-3.0.0`, `iree-3.0.0rc20241118`, `iree-3.0.0rc20241117`, `iree-3.0.0rc20241116`

![image](https://github.com/user-attachments/assets/d2f4f3d9-1c60-44bc-a0d2-21fb4eddcbce)

Reply:
That's pretty cool, thanks for sharing. Since this change doesn't need any updates to the iree requirements, I will leave it as is and change the requirements with the other PR tomorrow.

Reply:
Actually @raikonenfnu it turns out that on CDNA2, if I use the old version of iree I get a failure with attention. But with the latest, it is passing. So I will keep this for now and we can investigate further.

Reply:
that is odd though, I wasn't getting any attention failures on CDNA2, when working on the PR yesterday and this morning..